docker-new/Makefile

```diff
@@ -0,0 +1,136 @@
+# does not work very well if the path contains spaces in the name; in
+# fact lots of things break very badly; set this explicitly
+SCRIPT_DIR ?= ${PDO_SOURCE_ROOT}/docker-new
+
+DOCKER_USERNAME = $(LOGNAME)
+DOCKER_BUILDARGS += --build-arg PDO_HOSTNAME=$(PDO_HOSTNAME)
+DOCKER_BUILDARGS += --build-arg PDO_LEDGER_URL=$(PDO_LEDGER_URL)
+DOCKER_BUILDARGS += --build-arg UID=$(PDO_USER_UID)
+DOCKER_BUILDARGS += --build-arg GID=$(PDO_GROUP_UID)
+DOCKER_ARGS = --progress=plain $(DOCKER_BUILDARGS)
+
+IMAGES=base client services_base services ccf_base ccf
+
+# for the most part this is just used to force rebuild when the
+# PDO repository has changed
+TIMESTAMP := $(shell /bin/date "+%Y%m%d%H%M%S")
+
+all : $(addprefix build_,$(IMAGES))
+
+rebuild_% : repository
+	@ docker build $(DOCKER_ARGS) --build-arg REBUILD=$(TIMESTAMP) --tag pdo_$* \
+		--file $(SCRIPT_DIR)/pdo_$*.dockerfile .
+
+build_% : repository
+	@ docker build $(DOCKER_ARGS) --tag pdo_$* \
+		--file $(SCRIPT_DIR)/pdo_$*.dockerfile .
+
+clean_% :
+	@ docker rmi -f pdo_$*
+
+run_ccf : build_ccf
+	- docker rm -f ccf_container
+	@ docker run -v $(SCRIPT_DIR)/xfer/:/project/pdo/xfer --network host --name ccf_container -P -d pdo_ccf
+
+run_services : build_base build_services_base build_services
+	- docker rm -f services_container
+	@ docker run -v $(SCRIPT_DIR)/xfer/:/project/pdo/xfer --network host --name services_container -P -d pdo_services
+
+run_client : build_base build_client
+	- docker rm -f client_container
+	@ docker run -v $(SCRIPT_DIR)/xfer/:/project/pdo/xfer -it --network host --name client_container pdo_client
+
+# -----------------------------------------------------------------
+# We need a repository with the source for the branch we are going
+# to build. In theory this could just be a copy of the local source
+# tree but it might contain other artifacts. So instead we clone
+# the repository with just a single branch in it. The storage and
+# performance requirements are relatively low.
+# -----------------------------------------------------------------
+repository :
+	@ git clone --single-branch --branch $(PDO_BRANCH) --recurse-submodules $(PDO_REPO) repository
+
+clean_repository :
+	@ rm -rf repository
+
+# -----------------------------------------------------------------
+# Testing uses docker-compose to build a set of images that will be configured
+# to run with localhost as the interface. in order to get a clean shutdown we
+# don't use a docker-compose specification for the base, ccf_base and services_base
+# containers. since we are rebuilding the repository, we force recompilation of these
+# images from the rebuilt repository. the corresponding images are built using docker
+# build instead of being included in the docker-compose.
+# -----------------------------------------------------------------
+TEST_FILES = -f client_base.yaml
+TEST_FILES += -f services_base.yaml
+TEST_FILES += -f ccf_base.yaml
+TEST_FILES += -f test.yaml
+
```

Comment:
actually you are. we are testing pdo. we are not testing the other start up scripts, true. there is nothing there to test in the CI because you aren't running persistent services on user-specified interfaces.